src/main.rs

```diff
@@ -0,0 +1,88 @@
+#[subxt::subxt(runtime_metadata_path = "./metadata/kusama.scale")]
+pub mod kusama {
+    #[subxt(substitute_type = "sp_runtime::multiaddress::MultiAddress")]
+    use ::subxt::ext::sp_runtime::MultiAddress;
+}
+
+use anyhow::Result;
+use kusama::runtime_types::{
+    kusama_runtime::RuntimeCall, pallet_collective::pallet::Call as TechnicalCommitteeCall,
+    pallet_ranked_collective::pallet::Call as CollectiveCall,
+    pallet_utility::pallet::Call as UtilityCall,
+};
+use parity_scale_codec::Encode as _;
+use std::io::BufRead;
+use std::{fs, io, str::FromStr as _};
+use subxt::ext::sp_runtime::{AccountId32, MultiAddress};
+
+fn main() -> Result<()> {
+    let members = load_members()?;
+    assert_eq!(45, members.len());
+
+    let mut calls = Vec::new();
+
+    for member in members.iter() {
+        if member.name == "gavofyork" {
+            // gav has already been added
+            continue;
+        }
+        calls.push(RuntimeCall::FellowshipCollective(
+            CollectiveCall::add_member {
+                who: MultiAddress::Id(member.account_id.clone()),
+            },
+        ));
+        for rank in 0..member.rank {
+            // referendum required to promote to rank VII and higher
+            if rank < 6 {
+                calls.push(RuntimeCall::FellowshipCollective(
+                    CollectiveCall::promote_member {
+                        who: MultiAddress::Id(member.account_id.clone()),
+                    },
+                ))
+            }
+        }
+    }
+
+    let proposal = RuntimeCall::Utility(UtilityCall::batch { calls });
+    let length_bound = proposal.encoded_size() as u32;
```

Comment:
Wasn't sure what value to put for the `length_bound`, looking at the [code](https://github.com/paritytech/substrate/blob/17c07af0b953b84dbe89341294e98e586f9b4591/frame/collective/src/lib.rs#L772) it looks like it checks whether the encoded size of the proposal is <= this length bound

Reply:
IIUC it just needs an upper bound to calculate the worst case weight in advance.